system/Commands/Utilities/Optimize.php

```diff
@@ -99,32 +115,73 @@ private function removeFile(string $cache): void
         }
     }
 
-    private function enableCaching(): void
+    private function enableCaching(?bool $enableConfigCache, ?bool $enableLocatorCache, ?bool $restoreDefaults): void
     {
         $publisher = new Publisher(APPPATH, APPPATH);
 
         $config = APPPATH . 'Config/Optimize.php';
 
-        $result = $publisher->replace(
-            $config,
-            [
-                'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
-                'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
-            ]
-        );
+        // Prepare search and replace mappings
+        $searchReplace = [];
+
+        if ($restoreDefaults === true) {
+            $searchReplace = [
+                'public bool $configCacheEnabled = true;'  => 'public bool $configCacheEnabled = false;',
+                'public bool $locatorCacheEnabled = true;' => 'public bool $locatorCacheEnabled = false;',
+            ];
+        } else {
+            if ($enableConfigCache === true) {
+                $searchReplace['public bool $configCacheEnabled = false;'] = 'public bool $configCacheEnabled = true;';
+            }
 
-        if ($result) {
-            CLI::write(
-                'Config Caching and FileLocator Caching are enabled in "app/Config/Optimize.php".',
-                'green'
-            );
+            if ($enableLocatorCache === true) {
+                $searchReplace['public bool $locatorCacheEnabled = false;'] = 'public bool $locatorCacheEnabled = true;';
+            }
 
-            return;
+            // If no options provided, update both
+            if ($enableConfigCache === null && $enableLocatorCache === null) {
+                $searchReplace = [
+                    'public bool $configCacheEnabled = false;'  => 'public bool $configCacheEnabled = true;',
+                    'public bool $locatorCacheEnabled = false;' => 'public bool $locatorCacheEnabled = true;',
+                ];
+            }
         }
```

Comment:
if -r is present then it will take priority and ignore other options.

Reply:
Is it intentional?

Reply:
> Is it intentional?

Yes, it is.

Reply:
Okay.